00_Base/src/broadcaster/session.broadcaster.ts

```diff
@@ -22,34 +25,99 @@ export class SessionBroadcaster extends BaseBroadcaster {
+        }
+      },
     );
   }
 
-  private async broadcast(transactions: Transaction[]) {
+  private async broadcastPatch(
+    transactionsEvents: TransactionEvent[],
+  ): Promise<void> {
+    const transactions: Transaction[] = [];
+    for (const transactionsEvent of transactionsEvents) {
+      if (
+        transactionsEvent.meterValue &&
+        transactionsEvent.meterValue.length > 0 &&
+        transactionsEvent.triggerReason !== TriggerReasonEnumType.RemoteStart // skip remote start because it will be handled by PUT
+      ) {
+        // skip if there are no meter values in transaction event
+        const transaction: Transaction = (await transactionsEvent.$get(
+          'transaction',
+        )) as Transaction; // todo use Props
+        if (transaction) {
+          transaction.setDataValue('transactionEvents', [transactionsEvent]); // todo use Props
+          transaction.setDataValue('meterValues', transactionsEvent.meterValue); // todo use Props
+          transaction.transactionEvents = [transactionsEvent];
+          transaction.meterValues = transactionsEvent.meterValue;
+          transactions.push(transaction);
+        }
+      }
+    }
+    if (transactions.length === 0) {
+      return Promise.resolve(); // skipping because no transaction not yet created for transaction event
+    } else {
+      return await this.broadcast(transactions, true);
+    }
+  }
+
+  private async broadcast(transactions: Transaction[], isPatch = false) {
     // todo do we know if we can do put vs patch here, is there a way to have a delta?
     const sessions: Session[] =
       await this.sessionMapper.mapTransactionsToSessions(transactions);
 
     for (const session of sessions) {
-      await this.sendSessionToClients(session);
+      await this.sendSessionToClients(session, isPatch);
     }
   }
 
-  private async sendSessionToClients(session: Session): Promise<void> {
-    const params = PutSessionParams.build(session.id, session);
+  private async sendSessionToClients(
+    session: Session,
+    isPatch = false,
+  ): Promise<void> {
+    let params: any = PutSessionParams.build(session.id, session);
+    let clientApiRequest: any = this.sessionsClientApi.putSession;
+    if (isPatch) {
+      params = PatchSessionParams.build(
+        session.id,
+        this.getSessionForPatch(session),
+      );
+      clientApiRequest = this.sessionsClientApi.patchSession;
+    }
     const cpoCountryCode = session.country_code;
     const cpoPartyId = session.party_id;
     await this.sessionsClientApi.broadcastToClients(
       cpoCountryCode,
       cpoPartyId,
       ModuleId.Sessions,
       params,
-      this.sessionsClientApi.putSession.bind(this.sessionsClientApi),
+      clientApiRequest.bind(this.sessionsClientApi),
     );
   }
+
+  private getSessionForPatch(session: Session): Partial<Session> {
+    return {
+      kwh: session.kwh,
+      charging_periods: session.charging_periods,
```

Comment:
So we'd be sending the full charging periods? If so, seems to defeat the purpose of the patch.

Reply:
it is my understanding that we will be sending only the charging periods that come from the meter values that are in the transaction event so it should only be the latest charging periods right?